File: TeachingManagementPlatform.Api/Models/StudentList.cs
namespace TeachingManagementPlatform.Api.Models;

public class StudentList
{
    public int Id { get; set; }
    public int ClassId { get; set; }
    public string Name { get; set; } = string.Empty;
    public bool IsMain { get; set; }
    public DateTime CreatedAt { get; set; }

    // Navigation properties
    public Class Class { get; set; } = null!;
    public ICollection<StudentListColumn> Columns { get; set; } = new List<StudentListColumn>();
    public ICollection<StudentEntry> Entries { get; set; } = new List<StudentEntry>();
}

File: TeachingManagementPlatform.Frontend/src/types/studentList.ts
export interface StudentListColumn {
  id: number;
  studentListId: number;
  name: string;
  sortOrder: number;
}

export interface StudentEntry {
  id: number;
  studentListId: number;
  data: Record<string, string>;
  sortOrder: number;
}

export interface StudentList {
  id: number;
  classId: number;
  name: string;
  isMain: boolean;
  createdAt: string;
  columns: StudentListColumn[];
  entries: StudentEntry[];
}

export interface CreateStudentListRequest {
  name: string;
}

export interface UpdateStudentListRequest {
  name?: string;
}

export interface AddColumnRequest {
  name: string;
  sortOrder: number;
}

export interface AddStudentEntryRequest {
  data: Record<string, string>;
  sortOrder: number;
}

export interface UpdateStudentEntryRequest {
  data: Record<string, string>;
  sortOrder?: number;
}
